test/spec/features/completer/completer_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe 'complete' do
+
+  it 'outputs subcommand tree with --subcommand-tree' do
+    k = run 'kontena complete --subcommand-tree'
+    expect(k.code).to eq(0)
+    rows = k.out.split(/[\r\n]+/)
+    expect(rows).to include "kontena grid show"
+    expect(rows).to include "kontena stack install"
+    expect(rows).to include "kontena service restart"
+    expect(rows).to include "kontena volume list"
+    expect(rows).to include "kontena master user role add"
+    expect(rows).not_to include "kontena master user role add foo"
+    expect(rows.size > 100).to be_truthy
+  end
+
+  it 'can complete subcommands' do
+    k = run 'kontena complete kontena'
+    rows = k.out.split(/[\r\n]+/)
+    expect(rows).to include "stack"
+    expect(rows).to include "vault"
+    expect(rows).to include "whoami"
+  end
+
+  it 'can complete subsubcommands' do
+    k = run 'kontena complete kontena stack'
+    rows = k.out.split(/[\r\n]+/)
+    expect(rows).to include "install"
+    expect(rows).to include "deploy"
+  end
+
+  it 'can complete subsubsubcommands' do
+    k = run 'kontena complete kontena master user'
+    rows = k.out.split(/[\r\n]+/)
+    expect(rows).to include "invite"
+    expect(rows).to include "role"
+  end
+
+  it 'can complete subsubsubsubcommands' do
+    k = run 'kontena complete kontena master user role'
+    rows = k.out.split(/[\r\n]+/)
+    expect(rows).to include "add"
+    expect(rows).to include "remove"
+  end
+
+  it 'can complete master names' do
+    k = run 'kontena complete kontena master use'
+    expect(k.out).to match(/e2e/)
```

Comment:
It's not 100% guaranteed that there is a master with that name. I'm executing the e2e specs against random masters at least.